linetools/isgm/io.py

```diff
@@ -41,3 +50,67 @@ def abssys_from_json(filename):
 
     # Return
     return abs_sys
+
+
+def read_joebvp(filename, coord, llist=None):
+    """
+    Parameters
+    ----------
+    filename : str
+      joeB VP filename
+    coord : SkyCoord
+    llist : LineList, optional
+
+    Returns
+    -------
+    comps : list
+      list of AbsComponent objects
+    """
+    # init
+    if llist is None:
+        llist = LineList('ISM')
+    comps = []
+    # Read
+    vp_data = Table.read(filename, format='ascii')
+    nflags, nidx = np.unique(vp_data['nflag'], return_index=True)
+
+    # Subset by nflag; Build components
+    for nflag in nflags:
+        mt_lines = np.where(vp_data['nflag'] == nflag)[0]
+        if len(np.unique(vp_data['vel'][mt_lines])) != 1:
+            pdb.set_trace()
+        z_fit = vp_data['zsys'][mt_lines[0]] + vp_data['vel'][mt_lines[0]] * (1 + vp_data['zsys'][mt_lines[0]]) / ckms
+        # Loop on abs lines
+        alines = []
+        for idx in mt_lines:
+            zlim = [vp_data['zsys'][idx] +
+                    vp_data[vkey][idx] * (1 + vp_data['zsys'][idx]) / ckms
+                    for vkey in ['vlim1', 'vlim2']]
+            absline = AbsLine(vp_data['restwave'][idx] * u.AA, z=z_fit, zlim=zlim, linelist=llist)
+            # Add measurements [JB -- Want to capture anything else??]
+            absline.attrib['coord'] = coord
+            absline.attrib['flag_N'] = 1
+            absline.attrib['logN'] = vp_data['col'][idx]
+            absline.attrib['sig_logN'] = vp_data['sigcol'][idx]
+            absline.attrib['b'] = vp_data['bval'][idx]
+            absline.attrib['sig_b'] = vp_data['sigbval'][idx]
+            absline.attrib['z'] = z_fit
+            absline.attrib['sig_z'] = (1 + vp_data['z_comp'][idx]) * vp_data['sigvel'][idx] / ckms
+            absline.attrib['specfile'] = vp_data['specfile'][idx]
```

Comment:
What's in the 'specfile' is probably not going to be especially useful, since it records the path to file at the time of the fit.  What *is* useful (and even necessary for some of my analysis) is setting absline.analy['spec'].   For this reason, the function should allow the user to specify the spectrum, either an XSpectrum1D object or a path to the file.

Reply:
handled